packages/itwin/tree-widget/src/tree-widget-react/components/trees/models-tree/internal/FilteredTree.ts

```diff
@@ -39,30 +40,40 @@ interface ElementFilteredTreeNode extends BaseFilteredTreeNode {
 
 type FilteredTreeNode = GenericFilteredTreeNode | CategoryFilteredTreeNode | ElementFilteredTreeNode;
 
-export interface FilteredTree {
-  getVisibilityChangeTargets(node: HierarchyNode): VisibilityChangeTargets;
-}
-
-type CategoryKey = `${ModelId}-${CategoryId}`;
+type ModelCategoryKey = `${ModelId}-${CategoryId}`;
 
-function createCategoryKey(modelId: Id64String, categoryId: Id64String): CategoryKey {
+function createModelCategoryKey(modelId: Id64String, categoryId: Id64String): ModelCategoryKey {
   return `${modelId}-${categoryId}`;
 }
 
-/** @internal */
-export function parseCategoryKey(key: CategoryKey) {
+function parseModelCategoryKey(key: ModelCategoryKey) {
   const [modelId, categoryId] = key.split("-");
   return { modelId, categoryId };
 }
 
-interface VisibilityChangeTargets {
+interface FilterTargetsInternal {
   subjectIds?: Id64Set;
   modelIds?: Id64Set;
-  categories?: Set<CategoryKey>;
-  elements?: Map<CategoryKey, Set<ElementId>>;
+  categories?: Map<ModelId, Set<CategoryId>>;
+  elements?: Map<ModelCategoryKey, Set<ElementId>>;
 }
```

Comment:
Could we define these internal types, that are used only in one or two internal functions somewhere in the source file, somewhere next to where they're used?

In this case, maybe it could even be a class with `add` and `collect` methods?